packages/gatsby-plugin-sitemap/src/__tests__/gatsby-node.js

```diff
@@ -89,22 +86,29 @@ describe(`Test plugin sitemap`, () => {
         } 
     }`
     const options = {
-      output: `custom-sitemap.xml`,
-      serialize: ({ site, allSitePage }) =>
-        allSitePage.edges.map(edge => {
-          return {
-            url: site.siteMetadata.siteUrl + `/post` + edge.node.path,
-            changefreq: `weekly`,
-            priority: 0.8,
-          }
-        }),
-      exclude: [`/post/exclude-page`],
+      output: `custom-folder`,
+      resolveSiteUrl: () => siteUrl,
+      resolvePages: data => data.allSitePage.edges.map(edge => edge.node),
+      serialize: (page, { resolvePagePath }) => {
+        return {
+          url: resolvePagePath(page),
+          changefreq: `weekly`,
+          priority: 0.8,
+        }
+      },
+      excludes: [`/post/exclude-page`],
       query: customQuery,
     }
-    await onPostBuild({ graphql, pathPrefix }, options)
-    const [filePath, contents] = internals.writeFile.mock.calls[0]
-    expect(filePath).toEqual(path.join(`public`, `custom-sitemap.xml`))
-    expect(contents).toMatchSnapshot()
+
+    await onPostBuild({ graphql, pathPrefix, reporter }, options)
+
+    const {
+      destinationDir,
+      sourceData,
+    } = sitemap.simpleSitemapAndIndex.mock.calls[0][0]
+
+    expect(destinationDir).toEqual(path.join(`public`, `custom-folder`))
+    expect(sourceData).toMatchSnapshot()
     expect(graphql).toBeCalledWith(customQuery)
   })
   describe(`sitemap index`, () => {
```

Comment:
I don't think we need this describe block, we can move all of it to an it clause.